### PR TITLE
feat/get products

### PR DESCRIPTION
**Pull request by mufidat3250**
## What does the PR do?

 Developed an API endpoint to list products with pagination support.

- [x] get all product.
- [x] product does not exist.
- [x] products pagination

How should this be manually tested?

1. Make a Get request to the `\api\v1\products\?limit=10&page=1` endpoint with the following payload:

```json
Authorization: Bearer <access_token>

```
Successful Response

On a successful product list, the API should return a 200 get status code.

```json
{
  "success": true,
  "message": "Product retrieved successfully",
  "products": [
    {
      "name": "Product 1",
      "price": 29.99,
      "category": "Electronics"
    },
    {
      "name": "Product 2",
      "price": 19.99,
      "category": "Books"
    },
    ...
  ],
  "pagination": {
    "totalItems": 100,
    "totalPages": 10,
    "currentPage": 1
  },
"status_code": 200
}
```
success

Authentication Error:

When the current user is not properly authenticated to change the organization detail:

```json
{
  "status": "error",
  "message": "User is currently unauthorized, kindly authenticate to continue",
  "status_code": 401
}
```
Task

- [x] Make a get request for product.
- [x] Make a pagination query
Reference Issue: [Link to Issue](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/97)

### CODE REVIEW

**Comment by Heba-WebDev** [comment]
Change the title of the pr to feat: <name>

**Comment by mufidat3250** [comment]
> Change the title of the pr to feat:
done

File: db/migrations/1721835107794-migration.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class Migration1721835107794 implements MigrationInterface {
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+    }
+
+    public async down(queryRunner: QueryRunner): Promise<void> {
```
[the remark this answers is not in the document]
Comment (mufidat3250) [reply db/migrations/1721835107794-migration.ts]: change effected

**Comment by bamo100** [comment]
@mufidat3250 @davidolanrewaju Does this issue still prevails for fetching products tickets?

**Pull request closed without merging**